lib/openscap_parser/profiles.rb

```diff
@@ -1,25 +1,21 @@
 # frozen_string_literal: true
-require 'openscap_parser/test_result'
 
 module OpenscapParser
   # Methods related to saving profiles and finding which hosts
   # they belong to
   module Profiles
-    include TestResult
-
     def self.included(base)
       base.class_eval do
         def profiles
-          @profiles ||= {
-            profile_node['id'] => profile_node.at_css('title').text
-          }
-        end
+          @profiles ||= profile_nodes.inject({}) do |profiles, profile_node|
+            profiles[profile_node['id']] = profile_node.at_css('title').text
 
-        private
```

Comment:
Nothing in here needs to be private IMO. It's useful to retrieve the raw `Nokogiri::NodeSet` or an array of `OpenscapParser::Profile` objects.